# Add whole word mask support for lm fine-tune (#7925)
huggingface/transformers: examples/language-modeling/README.md

We use the `--mlm` flag so that the script may change its loss function.

> If use whole-word masking, use both `--mlm` and `--wwm` flag(for English Model).

Reviewer: ```suggestion
If using whole-word masking, use both the`--mlm` and `--wwm` flags.
```

```bash
export TRAIN_FILE=/path/to/dataset/wiki.train.raw
export TEST_FILE=/path/to/dataset/wiki.test.raw

python run_language_modeling.py \
    --output_dir=output \
    --model_type=roberta \
    --model_name_or_path=roberta-base \
    --do_train \
    --train_data_file=$TRAIN_FILE \
    --do_eval \
    --eval_data_file=$TEST_FILE \
    --mlm \
    --wwm
```

> For Chinese Model, we need to generate ref files, case it's char level.

Reviewer: ```suggestion
For Chinese models, we need to generate a reference files, case it's char level.
```
I don't understand the end of the comment: "case it's char level." Could you explain a bit more?

```bash
export TRAIN_FILE=/path/to/dataset/wiki.train.raw
export LTP_RESOURCE=/path/to/ltp/tokenizer
export BERT_RESOURCE=/path/to/bert/tokenizer
export SAVE_PATH=/path/to/data/ref.txt

python chinese_ref.py \
    --file_name=$TRAIN_FILE \
    --ltp=$LTP_RESOURCE
    --bert=$BERT_RESOURCE \
    --save_path=$SAVE_PATH 
```
Then: 

```bash
export TRAIN_FILE=/path/to/dataset/wiki.train.raw
export TEST_FILE=/path/to/dataset/wiki.test.raw
export REF_FILE=/path/to/ref.txt

> python run_language_modeling.py \

Reviewer: Can we leave one version with just mlm and no  wwm first?

Author: Sure, I update my readme, it's same with English version if only mlm.
